src/config/rcxml.c

```diff
@@ -1274,135 +1281,123 @@ entry(xmlNode *node, char *nodename, char *content)
 		} else {
 			wlr_log(WLR_ERROR, "Missing 'button' argument for tablet button mapping");
 		}
-	} else if (!strcasecmp(nodename, "motion.tabletTool")) {
+	} else if (!strcasecmp(name, "motion.tabletTool")) {
 		rc.tablet_tool.motion = tablet_parse_motion(content);
-	} else if (!strcasecmp(nodename, "relativeMotionSensitivity.tabletTool")) {
+	} else if (!strcasecmp(name, "relativeMotionSensitivity.tabletTool")) {
 		rc.tablet_tool.relative_motion_sensitivity =
 			tablet_get_dbl_if_positive(content, "relativeMotionSensitivity");
-	} else if (!strcasecmp(nodename, "ignoreButtonReleasePeriod.menu")) {
+	} else if (!strcasecmp(name, "ignoreButtonReleasePeriod.menu")) {
 		rc.menu_ignore_button_release_period = atoi(content);
-	} else if (!strcasecmp(nodename, "width.magnifier")) {
+	} else if (!strcasecmp(name, "width.magnifier")) {
 		rc.mag_width = atoi(content);
-	} else if (!strcasecmp(nodename, "height.magnifier")) {
+	} else if (!strcasecmp(name, "height.magnifier")) {
 		rc.mag_height = atoi(content);
-	} else if (!strcasecmp(nodename, "initScale.magnifier")) {
+	} else if (!strcasecmp(name, "initScale.magnifier")) {
 		set_float(content, &rc.mag_scale);
-	} else if (!strcasecmp(nodename, "increment.magnifier")) {
+	} else if (!strcasecmp(name, "increment.magnifier")) {
 		set_float(content, &rc.mag_increment);
-	} else if (!strcasecmp(nodename, "useFilter.magnifier")) {
+	} else if (!strcasecmp(name, "useFilter.magnifier")) {
 		set_bool(content, &rc.mag_filter);
 	}
 }
 
+static void xml_tree_walk(xmlNode *node, struct parser_state *state);
+
 static void
-process_node(xmlNode *node)
+traverse(xmlNode *node, struct parser_state *state)
 {
-	char *content;
-	static char buffer[256];
-	char *name;
-
-	content = (char *)node->content;
 	if (xmlIsBlankNode(node)) {
 		return;
 	}
-	name = nodename(node, buffer, sizeof(buffer));
-	entry(node, name, content);
-}
 
-static void xml_tree_walk(xmlNode *node);
-
```

Comment:
The only logical difference I see is that `xmlIsBlankNode()` used to guard only `entry()` and now guards `traverse()`. If a 'blank node' is ensured to not have properties nor children I think there is no logical difference.

Other than that, the `nodename` used to be built in the `process_node()` wrapper for `entry()` and is now built in `entry()` itself and the `process_node()` wrapper is removed.

I might have missed something though.